vcap_utils/vcap_utils/backends/base_tensorrt.py

```diff
@@ -0,0 +1,278 @@
+import numpy as np
+
+import pycuda.driver as cuda
+import tensorrt as trt
+import cv2
+
+from typing import Dict, List, Tuple, Optional, Any
+
+from vcap import (
+    Resize,
+    DETECTION_NODE_TYPE,
+    OPTION_TYPE,
+    BaseStreamState,
+    BaseBackend,
+    rect_to_coords,
+    DetectionNode,
+)
+
+
+class HostDeviceMem(object):
+    def __init__(self, host_mem, device_mem):
+        self.host = host_mem
+        self.device = device_mem
+
+    def __str__(self):
+        return "Host:\n" + str(self.host) + "\nDevice:\n" + str(self.device)
+
+    def __repr__(self):
+        return self.__str__()
+
+
+class AllocatedBuffer:
+    def __init__(self, inputs_, outputs_, bindings_, stream_):
+        self.inputs = inputs_
+        self.outputs = outputs_
+        self.bindings = bindings_
+        self.stream = stream_
+
+
+class BaseTensorRTBackend(BaseBackend):
+    def __init__(self, engine_bytes, width, height, device_id):
+        super().__init__()
+        gpu_devide_id = int(device_id[4:])
+        cuda.init()
+        dev = cuda.Device(gpu_devide_id)
+        self.ctx = dev.make_context()
+        TRT_LOGGER = trt.Logger(trt.Logger.WARNING)
```

Comment:
How many logs are there if logger is set to INFO? We may want to pull this setting from an environment variable

Reply:
Not much. But it's necessary when creating trt runtim

Reply:
Actually, I didn't see any logs.

Reply:
Actually, I didn't see any logs, even change it to INFO

Reply:
Okay, in that case, keep it at INFO. I'd rather not miss logs, and if too many logs becomes a problem we'll add an environment variable for filtering them.

Reply:
Okay, in that case, keep it at INFO. I'd rather not miss logs, and if too many logs becomes a problem we'll add an environment variable for filtering them.